src/main/java/org/opensearch/ad/ml/CheckpointDao.java

```diff
@@ -298,16 +332,39 @@ public String toCheckpoint(EntityModel model) {
         });
     }
 
-    /**
-     * Deletes the model checkpoint for the id.
-     *
-     * @deprecated use deleteModelCheckpoint with listener instead
-     *
-     * @param modelId ID of the model checkpoint
-     */
-    @Deprecated
-    public void deleteModelCheckpoint(String modelId) {
-        clientUtil.<DeleteRequest, DeleteResponse>timedRequest(new DeleteRequest(indexName, modelId), logger, client::delete);
+    private String rcfModelToCheckpoint(RandomCutForest model) {
+        LinkedBuffer borrowedBuffer = null;
+        try {
+            borrowedBuffer = deserializeRCFBufferPool.borrowObject();
+            try {
+                return rcfModelToCheckpoint(model, borrowedBuffer);
+            } catch (Exception e) {
+                if (borrowedBuffer != null) {
+                    deserializeRCFBufferPool.invalidateObject(borrowedBuffer);
+                    borrowedBuffer = null;
+                }
+            } finally {
+                if (borrowedBuffer != null) {
+                    deserializeRCFBufferPool.returnObject(borrowedBuffer);
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Failed to borrow an buffer from object pool", e);
+            return rcfModelToCheckpoint(model, null);
+        }
+        return null;
+    }
+
+    String rcfModelToCheckpoint(RandomCutForest model, LinkedBuffer buffer) {
+        final LinkedBuffer deserializationBuffer = buffer == null ? LinkedBuffer.allocate(deserializeRCFBufferSize) : buffer;
```

Comment:
Why are we calling it deserializationBuffer if we are serializing it in this function?

Reply:
good catch. Will change to serializationBuffer